### DISCUSSION TITLE
Set value for field type

### DISCUSSION THREAD

**Discussion by stephenr85** [Laravel-Backpack/community-forum#592] [General] [answered]
I'm using [schemaless-attributes](https://github.com/spatie/laravel-schemaless-attributes) for my JSON columns, but the fields always want an array. Is there a method in the field API that lets you manipulate the value that's fed to the field?

If not, is the recommended way to do this to create a separate attribute cast (i.e. meta_array) on the model, and point the field at that?

Semi-related...what if I just want to throw a JSON blob into a text area?

**Comment by stephenr85** [comment]
In the meantime, I'm using the attribute cast, for anyone facing the same issue with a custom class cast:

```php


    public function modelMetaArray() : Attribute
    {
        return Attribute::make(
            get: fn($value) => $this->meta->toArray(),
            set: fn($value) => ['meta' => $value],
        );
    }
```

**Reply by christmex** [reply, marked as answer]
Hi @stephenr85 in my opinion I think I would go the same way as you, it's already a good way.

<img width="607" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/34405837/53b7c7cb-97a8-469a-9e40-620adf7b215a">

Backpack PRO also has this field called table to manage the same as your case.
